core/src/apps/zcash/signer.py

```diff
@@ -44,8 +50,25 @@ def __init__(
         if tx.version != 5:
             raise DataError("Expected transaction version 5.")
 
+        assert approver is None
+        approver = ZcashApprover(tx, coin)
+
         super().__init__(tx, keychain, coin, approver)
 
+        if ZCASH_SHIELDED:
```

Comment:
8384dc6e58fa81854af6590e4b2fd37615993d94

Reply:
I added three conditions into `sanitize_sign_tx`. First, I added 
```
tx.inputs_count == 0 or tx.orchard_params.inputs_count == 0
```
condition to forbid spending from transparent and Orchard account simultaneously. Second, I added
```
tx.orchard_params.inputs_count + tx.orchard_params.outputs_count > 0
```
condition to reject empty Orchard bundles. Finally, I added
```
utils.ZCASH_SHIELDED or tx.orchard params is None
```
condition, so I think that all `if ZCASH_SHIELDED and self.orchard is not None` can be turned into just `if self.orchard is not None`.

Reply:
correct me if i'm wrong, but we still need to trigger the `MISMATCH`?

otherwise I would:
1. disable safety checks
2. spend 0 transparent and N orchard inputs, `orchard_params.address_n = m/32'/144'/0'`
3. send to a destination address
4. send to a transparent change account at `address_n = m/32'/144'/0'/1/0`
5. because there were no transparent inputs, the match checker will accept this as the only attribute
6. ...and allow it as change address without showing it on screen
7. thus invisibly deanonymizing my funds

Reply:
> correct me if i'm wrong, but we still need to trigger the MISMATCH?

you are right

Reply:
always use `utils.ZCASH_SHIELDED`: 99e3384ea9527760de7be2e5a07e3ffc187a81f8

Reply:
set `MISMATCH`: 821d0e2cb1179d076e255c30eddc7d8a1e60d23c